packages/botonic-cli/tests/analytics/telemetry.test.ts

```diff
@@ -0,0 +1,85 @@
+import { assert } from 'console'
+
+import { Telemetry } from '../../src/analytics/telemetry'
+import { pathExists } from '../../src/util/file-system'
+
+process.env.BOTONIC_DISABLE_ANALYTICS = '1'
+assert(process.env.BOTONIC_DISABLE_ANALYTICS === '1')
+
+describe('TEST: Telemetry', () => {
+  let telemetry: Telemetry
+  beforeEach(() => {
+    telemetry = new Telemetry()
+  })
+
+  afterEach(() => {})
+  it('Initializes correctly Telemetry', () => {
+    expect(telemetry.isAnalyticsEnabled).toBe(false)
+    expect(telemetry.analyticsService).toBeTruthy()
+    expect(pathExists(telemetry.globalCredentialsHandler.homePath)).toBeTruthy()
+  })
+  it('Creates anonymousId if not exists', () => {
+    const currentInfo = telemetry.globalCredentialsHandler.load()
+    telemetry.globalCredentialsHandler.dump({
+      analytics: { anonymous_id: '' },
+    })
+    telemetry.globalCredentialsHandler.load()
+    assert(!telemetry.globalCredentialsHandler.getAnonymousId())
+    const anonymousId = telemetry.createAnonymousIdIfNotExists()
+    expect(anonymousId).toBeTruthy()
+    telemetry.globalCredentialsHandler.dump(currentInfo)
+  })
+  it('Generates the correct properties', () => {
+    const sut = Object.keys(
+      telemetry.getTelemetryProperties({ addedField: 'test' })
+    )
+    expect(sut).toEqual([
+      'platform',
+      'arch',
+      'timezone',
+      'timestamp',
+      'is_tty',
+      'framework_path',
+      'system_path',
+      'node_version',
+      'botonic_cli_version',
+      'botonic_dependencies',
+      'addedField',
+    ])
+  })
+  it('Do telemetry calls properly', () => {
+    const track = (Telemetry.prototype.track = jest.fn())
```

Comment:
it looks like your asserting the call on the same method that you call. am I wrong?

Reply:
taken from this example: https://stackoverflow.com/a/50108368, first do the assigment in the right parentheses, then assign into track variable.

Reply:
call me , I don't get it